contracts/external/cw-bounties/src/contract.rs

```diff
@@ -0,0 +1,306 @@
+    if sent_amount != amount.amount {
+        return Err(ContractError::InvalidAmount {
+            expected: amount.amount,
+            actual: sent_amount,
+        });
+    };
+
+    // Check bounty title is not empty string
+    if title.is_empty() {
+        return Err(ContractError::EmptyTitle {});
+    }
+
+    // Increment and get the next bounty ID
+    let id = ID.update(deps.storage, |mut id| -> StdResult<u64> {
+        id += 1;
+        Ok(id)
+    })?;
+
+    // Save the bounty
+    BOUNTIES.save(
+        deps.storage,
+        id,
+        &Bounty {
+            id,
+            amount,
+            title,
+            description,
+            status: BountyStatus::Open,
+            created_at: env.block.time.seconds(),
+            updated_at: None,
+        },
+    )?;
+
+    Ok(Response::default()
+        .add_attribute("action", "create_bounty")
+        .add_attribute("id", id.to_string()))
+}
+
+pub fn close(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    id: u64,
+) -> Result<Response, ContractError> {
+    // Check bounty exists
+    let mut bounty = BOUNTIES.load(deps.storage, id)?;
+
+    // Check bounty is open
+    if bounty.status != BountyStatus::Open {
+        return Err(ContractError::NotOpen {});
+    };
+
+    bounty.status = BountyStatus::Closed {
+        closed_at: env.block.time.seconds(),
+    };
+    BOUNTIES.save(deps.storage, id, &bounty)?;
+
+    // Pay out remaining funds to owner
+    // Only owner can call this, so sender is owner
+    let msg = BankMsg::Send {
+        to_address: info.sender.to_string(),
+        amount: vec![bounty.amount],
+    };
+
+    Ok(Response::default()
+        .add_message(msg)
+        .add_attribute("action", "close_bounty"))
+}
+
+pub fn pay_out(
+    deps: DepsMut,
+    env: Env,
+    id: u64,
+    recipient: String,
+) -> Result<Response, ContractError> {
+    // Check bounty exists
+    let mut bounty = BOUNTIES.load(deps.storage, id)?;
+
+    // Check bounty is open
+    if bounty.status != BountyStatus::Open {
+        return Err(ContractError::NotOpen {});
+    }
+
+    // Validate recipient address
+    deps.api.addr_validate(&recipient)?;
+
+    // Set bounty status to claimed
+    bounty.status = BountyStatus::Claimed {
+        claimed_by: recipient.clone(),
+        claimed_at: env.block.time.seconds(),
+    };
+    BOUNTIES.save(deps.storage, id, &bounty)?;
+
+    // Message to pay out remaining funds to recipient
+    let msg = BankMsg::Send {
+        to_address: recipient.clone(),
+        amount: vec![bounty.clone().amount],
+    };
+
+    Ok(Response::new()
+        .add_message(msg)
+        .add_attribute("action", "pay_out_bounty")
+        .add_attribute("bounty_id", id.to_string())
+        .add_attribute("amount", bounty.amount.to_string())
+        .add_attribute("recipient", recipient))
+}
+
+pub fn update(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    id: u64,
+    new_amount: Coin,
```

Comment:
as it also includes the denom i'd suggeest
```suggestion
    new_coin: Coin,
```